src/base/UniqueId.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+namespace NuTo
+{
+//! @brief This class provides a unique id (beginning at 0 and incrementing for each object).
+//! @tparam T Without the template, the id would be unique for all classes. Say, 1000 nodes are created first and get
+//! the ids 0...999. Element ids would then start at 1000. With this template, T=Node or T=Element, each T starts at 0
+//! again.
+template <typename T>
+class UniqueId
+{
+public:
+    UniqueId()
+        : mId(mIdCounter++)
+    {
+    }
+
+    int Id() const
+    {
```

Comment:
Not sure if this is what you intended.

It will return the `Id` of the **next** object that will eventually be created. Not the last one.

Mb this is what you wanted?
~~~cpp
private:
static int mIdCounter = -1;
public:
UniqueId() : mId(++mIdCounter){}
~~~

Reply:
Have a look at the test. First call to Id() returns 0.

Reply:
You are right. For some reason I thought it would return `mIdCounter`